{
    "id": "CVE-2024-44941",
    "sourceIdentifier": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
    "published": "2024-08-26T12:15:06.107",
    "lastModified": "2024-09-12T20:57:26.143",
    "vulnStatus": "Analyzed",
    "cveTags": [],
    "descriptions": [
        {
            "lang": "en",
            "value": "In the Linux kernel, the following vulnerability has been resolved:\n\nf2fs: fix to cover read extent cache access with lock\n\nsyzbot reports a f2fs bug as below:\n\nBUG: KASAN: slab-use-after-free in sanity_check_extent_cache+0x370\/0x410 fs\/f2fs\/extent_cache.c:46\nRead of size 4 at addr ffff8880739ab220 by task syz-executor200\/5097\n\nCPU: 0 PID: 5097 Comm: syz-executor200 Not tainted 6.9.0-rc6-syzkaller #0\nHardware name: Google Google Compute Engine\/Google Compute Engine, BIOS Google 03\/27\/2024\nCall Trace:\n <TASK>\n __dump_stack lib\/dump_stack.c:88 [inline]\n dump_stack_lvl+0x241\/0x360 lib\/dump_stack.c:114\n print_address_description mm\/kasan\/report.c:377 [inline]\n print_report+0x169\/0x550 mm\/kasan\/report.c:488\n kasan_report+0x143\/0x180 mm\/kasan\/report.c:601\n sanity_check_extent_cache+0x370\/0x410 fs\/f2fs\/extent_cache.c:46\n do_read_inode fs\/f2fs\/inode.c:509 [inline]\n f2fs_iget+0x33e1\/0x46e0 fs\/f2fs\/inode.c:560\n f2fs_nfs_get_inode+0x74\/0x100 fs\/f2fs\/super.c:3237\n generic_fh_to_dentry+0x9f\/0xf0 fs\/libfs.c:1413\n exportfs_decode_fh_raw+0x152\/0x5f0 fs\/exportfs\/expfs.c:444\n exportfs_decode_fh+0x3c\/0x80 fs\/exportfs\/expfs.c:584\n do_handle_to_path fs\/fhandle.c:155 [inline]\n handle_to_path fs\/fhandle.c:210 [inline]\n do_handle_open+0x495\/0x650 fs\/fhandle.c:226\n do_syscall_x64 arch\/x86\/entry\/common.c:52 [inline]\n do_syscall_64+0xf5\/0x240 arch\/x86\/entry\/common.c:83\n entry_SYSCALL_64_after_hwframe+0x77\/0x7f\n\nWe missed to cover sanity_check_extent_cache() w\/ extent cache lock,\nso, below race case may happen, result in use after free issue.\n\n- f2fs_iget\n - do_read_inode\n  - f2fs_init_read_extent_tree\n  : add largest extent entry in to cache\n\t\t\t\t\t- shrink\n\t\t\t\t\t - f2fs_shrink_read_extent_tree\n\t\t\t\t\t  - __shrink_extent_tree\n\t\t\t\t\t   - __detach_extent_node\n\t\t\t\t\t   : drop largest extent entry\n  - sanity_check_extent_cache\n  : access et->largest w\/o lock\n\nlet's refactor sanity_check_extent_cache() to avoid extent cache access\nand call it before f2fs_init_read_extent_tree() to fix this issue."
        },
        {
            "lang": "es",
            "value": "En el kernel de Linux, se resolvi\u00f3 la siguiente vulnerabilidad: f2fs: correcci\u00f3n para cubrir el acceso a la cach\u00e9 de extensi\u00f3n de lectura con bloqueo syzbot informa un error de f2fs como se muestra a continuaci\u00f3n: ERROR: KASAN: slab-use-after-free in sanity_check_extent_cache+0x370\/0x410 fs\/ f2fs\/extent_cache.c:46 Lectura de tama\u00f1o 4 en la direcci\u00f3n ffff8880739ab220 por tarea syz-executor200\/5097 CPU: 0 PID: 5097 Comm: syz-executor200 No contaminado 6.9.0-rc6-syzkaller #0 Nombre de hardware: Google Google Compute Engine \/Google Compute Engine, BIOS Google 27\/03\/2024 Seguimiento de llamadas:  __dump_stack lib\/dump_stack.c:88 [en l\u00ednea] dump_stack_lvl+0x241\/0x360 lib\/dump_stack.c:114 print_address_description mm\/kasan\/report.c: 377 [en l\u00ednea] print_report+0x169\/0x550 mm\/kasan\/report.c:488 kasan_report+0x143\/0x180 mm\/kasan\/report.c:601 sanity_check_extent_cache+0x370\/0x410 fs\/f2fs\/extent_cache.c:46 do_read_inode fs\/f2fs \/inode.c:509 [en l\u00ednea] f2fs_iget+0x33e1\/0x46e0 fs\/f2fs\/inode.c:560 f2fs_nfs_get_inode+0x74\/0x100 fs\/f2fs\/super.c:3237 generic_fh_to_dentry+0x9f\/0xf0 fs\/libfs.c:1413 exportar fs_decode_fh_raw +0x152\/0x5f0 fs\/exportfs\/expfs.c:444 exportfs_decode_fh+0x3c\/0x80 fs\/exportfs\/expfs.c:584 do_handle_to_path fs\/fhandle.c:155 [en l\u00ednea] handle_to_path fs\/fhandle.c:210 [en l\u00ednea] do_handle_open +0x495\/0x650 fs\/fhandle.c:226 do_syscall_x64 arch\/x86\/entry\/common.c:52 [en l\u00ednea] do_syscall_64+0xf5\/0x240 arch\/x86\/entry\/common.c:83 Entry_SYSCALL_64_after_hwframe+0x77\/0x7f Nos perdimos cubra sanity_check_extent_cache() con bloqueo de cach\u00e9 de extensi\u00f3n, por lo que, puede ocurrir el siguiente caso de ejecuci\u00f3n, lo que resultar\u00e1 en un problema de use after free. - f2fs_iget - do_read_inode - f2fs_init_read_extent_tree: agrega la entrada de mayor extensi\u00f3n al cach\u00e9 - encogimiento - f2fs_shrink_read_extent_tree - __shrink_extent_tree - __detach_extent_node: elimina la entrada de mayor extensi\u00f3n - sanity_check_extent_cache: accede a et-&gt;largest sin bloqueo, refactoricemos sanity_check_extent_cache() para evitar el cach\u00e9 de extensi\u00f3n acceso y ll\u00e1melo antes de f2fs_init_read_extent_tree() para solucionar este problema."
        }
    ],
    "metrics": {
        "cvssMetricV31": [
            {
                "source": "nvd@nist.gov",
                "type": "Primary",
                "cvssData": {
                    "version": "3.1",
                    "vectorString": "CVSS:3.1\/AV:L\/AC:L\/PR:L\/UI:N\/S:U\/C:H\/I:H\/A:H",
                    "baseScore": 7.8,
                    "baseSeverity": "HIGH",
                    "attackVector": "LOCAL",
                    "attackComplexity": "LOW",
                    "privilegesRequired": "LOW",
                    "userInteraction": "NONE",
                    "scope": "UNCHANGED",
                    "confidentialityImpact": "HIGH",
                    "integrityImpact": "HIGH",
                    "availabilityImpact": "HIGH"
                },
                "exploitabilityScore": 1.8,
                "impactScore": 5.9
            }
        ]
    },
    "weaknesses": [
        {
            "source": "nvd@nist.gov",
            "type": "Primary",
            "description": [
                {
                    "lang": "en",
                    "value": "CWE-416"
                }
            ]
        }
    ],
    "configurations": [
        {
            "nodes": [
                {
                    "operator": "OR",
                    "negate": false,
                    "cpeMatch": [
                        {
                            "vulnerable": true,
                            "criteria": "cpe:2.3:o:linux:linux_kernel:*:*:*:*:*:*:*:*",
                            "versionEndExcluding": "6.6.47",
                            "matchCriteriaId": "CF4F8EA7-C643-4B9C-8E32-BD81930A5E8B"
                        },
                        {
                            "vulnerable": true,
                            "criteria": "cpe:2.3:o:linux:linux_kernel:*:*:*:*:*:*:*:*",
                            "versionStartIncluding": "6.7",
                            "versionEndExcluding": "6.10.6",
                            "matchCriteriaId": "C1CDF130-CCA6-47F6-9718-8949BF4E05CC"
                        }
                    ]
                }
            ]
        }
    ],
    "references": [
        {
            "url": "https:\/\/git.kernel.org\/stable\/c\/263df78166d3a9609b97d28c34029bd01874cbb8",
            "source": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
            "tags": [
                "Patch"
            ]
        },
        {
            "url": "https:\/\/git.kernel.org\/stable\/c\/323ef20b5558b9d9fd10c1224327af6f11a8177d",
            "source": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
            "tags": [
                "Patch"
            ]
        },
        {
            "url": "https:\/\/git.kernel.org\/stable\/c\/d7409b05a64f212735f0d33f5f1602051a886eab",
            "source": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
            "tags": [
                "Patch"
            ]
        }
    ]
}